gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -168,14 +170,17 @@ public class GobblinServiceJobScheduler extends JobScheduler implements SpecCata
    * so only they would be loaded during DR handling.
    */
   public static final String DR_FILTER_TAG = "dr";
+  private final boolean dagProcessingEngineEnabled;
+  private final DagManagement dagManagement;
 
   @Inject
   public GobblinServiceJobScheduler(@Named(InjectionNames.SERVICE_NAME) String serviceName,
       Config config,
       Optional<HelixManager> helixManager, Optional<FlowCatalog> flowCatalog,
       Orchestrator orchestrator, SchedulerService schedulerService, Optional<UserQuotaManager> quotaManager, Optional<Logger> log,
       @Named(InjectionNames.WARM_STANDBY_ENABLED) boolean isWarmStandbyEnabled,
-      Optional<FlowTriggerHandler> flowTriggerHandler) throws Exception {
+      Optional<FlowTriggerHandler> flowTriggerHandler, DagManagement dagManagement,
```

Comment:
Yea, I actually wanted to create those objects regardless of the enabled flag so at least the object creation can be tested in prod.

Reply:
I'd whole-heartedly urge us to ONLY create required instances.  otherwise a not-yet-ready-for-prod instance that fails creation can bring down the entire application.  best practice is to guard significant new/reworked functionality behind config, and this includes initialization